lib/build-components/build_components/templates/python/Dockerfile.j2

```diff
@@ -0,0 +1,129 @@
+{% from "macros.j2" import install_poetry, build_component, collect_components, add_poetry, add_user, add_collected_components %}
```

Comment:
Can't remember if I've mentioned it before, but have we looked at using an alternate build system like bazel https://github.com/bazelbuild/rules_docker instead of adding a layer of templating on top of Dockerfiles before?  Obvs out of scope here, I just stumbled across it a while back looking into how distroless worked, and it seemed neat and potentially powerful and convenient for doing complicated things building docker images.

Reply:
Yeah, I remember you mentioning it. I took a brief look at it, but it didn't seem immediately applicable to our use case. I'm not opposed to adopting it though, if it can support our needs. Perhaps it warrants a closer look.